osu.Game/Overlays/Profile/Sections/DrawableHistoryItem.cs

```diff
@@ -99,33 +97,19 @@ private void handleString(string s)
             if (string.IsNullOrEmpty(s))
                 return;
 
-            if (!s.Contains(':'))
+            if (!s.Contains('{'))
             {
                 AddText(s);
                 return;
             }
 
-            var variableIndex = s.IndexOf(':');
+            var variableIndex = s.IndexOf('{');
+            var variableEndIndex = s.IndexOf('}');
+            var variableName = s.Substring(variableIndex + 1, variableEndIndex - variableIndex - 1);
 
             AddText(s.Substring(0, variableIndex));
-
-            var trimmed = handleVariable(s.Substring(variableIndex + 1));
-
-            handleString(trimmed);
-        }
-
-        private string handleVariable(string s)
-        {
-            foreach (var (name, action) in properties)
-            {
-                if (s.StartsWith(name))
-                {
-                    action.Invoke();
-                    return s.Substring(name.Length);
-                }
-            }
-
-            return s;
+            HandleVariable(variableName);
+            handleString(s.Substring(variableEndIndex + 1));
```

Comment:
Why recursion? This can be done iteratively. (And without creating string copies, no less.)